src/cargo/core/compiler/compilation.rs

```diff
@@ -75,26 +75,11 @@ pub struct Compilation<'cfg> {
 
 impl<'cfg> Compilation<'cfg> {
     pub fn new<'a>(bcx: &BuildContext<'a, 'cfg>) -> CargoResult<Compilation<'cfg>> {
-        // If we're using cargo as a rustc wrapper then we're in a situation
-        // like `cargo fix`. For now just disregard the `RUSTC_WRAPPER` env var
-        // (which is typically set to `sccache` for now). Eventually we'll
-        // probably want to implement `RUSTC_WRAPPER` for `cargo fix`, but we'll
-        // leave that open as a bug for now.
-        let mut rustc = if bcx.build_config.cargo_as_rustc_wrapper {
-            let mut rustc = bcx.rustc.process_no_wrapper();
-            let prog = rustc.get_program().to_owned();
-            rustc.env("RUSTC", prog);
-            rustc.program(env::current_exe()?);
-            rustc
-        } else {
-            bcx.rustc.process()
-        };
+        let mut rustc = bcx.rustc.process();
+
         if bcx.config.extra_verbose() {
             rustc.display_env_vars();
         }
-        for (k, v) in bcx.build_config.extra_rustc_env.iter() {
-            rustc.env(k, v);
-        }
         for arg in bcx.build_config.extra_rustc_args.iter() {
```

Comment:
wondering if this should be moved as well

Reply:
the extra_args specifically

Reply:
I think so yeah, because this is only applicable for cargo-fix. I think though if we do the `ProcessBuilder` suggestion from below we can delete all these various fields in `BuildConfig` and instead just have an `Option<ProcessBuilder>` perhaps?